src/lTerm_widget.mli

```diff
@@ -253,6 +256,31 @@ class ['a] radiobutton : 'a radiogroup -> string -> 'a -> object
 
 end
 
+(** {6 Scrollbars} *)
+
+(** methods required for a widget to supprt scrollbars *)
+class type scrollable = object
+  method full_size : LTerm_geom.size
+    (* full size of scrollable window *)
+
+  method offset : LTerm_geom.coord
+    (* offset within window *)
+  
+  method set_offset : LTerm_geom.coord -> unit
+    (* set offset within window *)
+
+end
+
+(** vertical scrollbar *)
+class vscrollbar : 
```

Comment:
You need to either add an extra `unit` argument or have the last argument non-labelled, otherwise you have to pass either none or all of the optional argument. To see the problem you can try adding `~scroll_bar_thickness:1` to the example